colour/colorimetry/spectrum.py

```diff
@@ -53,6 +56,88 @@
            'zeros_spd',
            'ones_spd']
 
+DEFAULT_WAVELENGTH_DECIMALS = 10
+"""
+Default wavelength precision decimals.
+
+DEFAULT_WAVELENGTH_DECIMALS : int
+"""
+
+
+class SpectralMapping(ArbitraryPrecisionMapping):
+    """
+    Defines the base mapping for spectral data.
+
+    It enables usage of floating point wavelengths as keys by rounding them at
+    a specfic decimals count.
+
+    Parameters
+    ----------
+    data : dict or SpectralMapping, optional
+        Spectral data in a *dict* or *SpectralMapping* as follows:
+        {wavelength :math:`\lambda_{i}`: value,
+        wavelength :math:`\lambda_{i+1}`,
+        ...,
+        wavelength :math:`\lambda_{i+n}`}
+    wavelength_decimals : int, optional
+        Decimals count the keys will be rounded at.
```

Comment:
Ah yeah, that's what lead me to believe it should work, Didn't follow with what `round` does with it.